Errors popped up for Visual Studio 16.10 Preview 3

I'm posting this because I'm not sure if it's a problem with Visual Studio or Boost.ut or if it's me.
## Expected Behavior
These worked in the 16.9.x versions of Visual Studio.
## Actual Behavior
Errors popped up for Visual Studio 16.10 Preview 3
## Steps to Reproduce the Problem

  1. Install Visual Studio 16.10 Preview 3
  1. Try to compile
  1. Error

## Specifications

  - Version: boost.ut trunk
  - Platform: Windows 10 x64
  - Subsystem: ?

```shell
====================[ Build | all | Debug Windows ]=============================
"C:\Program Files\JetBrains\CLion 2020.2\bin\cmake\win\bin\cmake.exe" --build D:\dev\ToolsLibrary\cmake-build-debug-windows --target all
Scanning dependencies of target ToolsLibrary_random_test
[  7%] Building CXX object tests/src/CMakeFiles/ToolsLibrary_random_test.dir/random_test.cpp.obj
random_test.cpp
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): error C3378: a declaration can be exported only from a module interface unit
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): error C2230: could not find module 'boost.ut'
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(10): error C3378: a declaration can be exported only from a module interface unit
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(10): error C7635: a module import declaration cannot appear in an 'export' declaration sequence
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(16): warning C5244: '#include <ciso646>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <ciso646>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(16): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(55): warning C5244: '#include <array>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <array>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(55): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(56): warning C5244: '#include <iostream>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <iostream>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(56): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(57): warning C5244: '#include <sstream>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <sstream>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(57): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(58): warning C5244: '#include <string_view>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <string_view>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(58): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(60): warning C5244: '#include <vector>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <vector>;'.
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(60): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(70): error C3378: a declaration can be exported only from a module interface unit
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(9): warning C5244: '#include <algorithm>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <algorithm>;'.
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(9): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(10): warning C5244: '#include <cassert>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <cassert>;'.
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(10): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(11): warning C5244: '#include <numeric>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <numeric>;'.
D:\dev\ToolsLibrary\src\tl/algorithm.hpp(11): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
D:\dev\ToolsLibrary\src\tl/random.hpp(8): warning C5244: '#include <random>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <random>;'.
D:\dev\ToolsLibrary\src\tl/random.hpp(8): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): note: see module 'boost.ut' declaration
NMAKE : fatal error U1077: 'C:\PROGRA~2\MICROS~1\2019\Preview\VC\Tools\MSVC\1429~1.300\bin\Hostx86\x86\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\bin\HostX86\x86\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\bin\HostX86\x86\nmake.exe"' : return code '0x2'
Stop.
```

if I enable /Wv:18 it gets rid of some of these lines.

```shell
====================[ Build | all | Debug Windows ]=============================
"C:\Program Files\JetBrains\CLion 2020.2\bin\cmake\win\bin\cmake.exe" --build D:\dev\ToolsLibrary\cmake-build-debug-windows --target all
[  7%] Building CXX object tests/src/CMakeFiles/ToolsLibrary_random_test.dir/random_test.cpp.obj
random_test.cpp
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): error C3378: a declaration can be exported only from a module interface unit
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(9): error C2230: could not find module 'boost.ut'
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(10): error C3378: a declaration can be exported only from a module interface unit
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(10): error C7635: a module import declaration cannot appear in an 'export' declaration sequence
D:\dev\ToolsLibrary\cmake-build-debug-windows\_deps\ut-src\include\boost/ut.hpp(70): error C3378: a declaration can be exported only from a module interface unit
NMAKE : fatal error U1077: 'C:\PROGRA~2\MICROS~1\2019\Preview\VC\Tools\MSVC\1429~1.300\bin\Hostx86\x86\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\bin\HostX86\x86\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\bin\HostX86\x86\nmake.exe"' : return code '0x2'
Stop.
```

if I switch `#include <boost/ut.hpp>` to `import boost.ut;` it's errors:
```shell
====================[ Build | all | Debug Windows ]=============================
"C:\Program Files\JetBrains\CLion 2020.2\bin\cmake\win\bin\cmake.exe" --build D:\dev\ToolsLibrary\cmake-build-debug-windows --target all
Scanning dependencies of target ToolsLibrary_random_test
[  7%] Building CXX object tests/src/CMakeFiles/ToolsLibrary_random_test.dir/random_test.cpp.obj
random_test.cpp
D:\dev\ToolsLibrary\tests\src\random_test.cpp(2): error C2230: could not find module 'boost.ut'
```

> if I switch `#include <boost/ut.hpp>` to `import boost.ut;` it's errors:
> 
> ```shell
> ====================[ Build | all | Debug Windows ]=============================
> "C:\Program Files\JetBrains\CLion 2020.2\bin\cmake\win\bin\cmake.exe" --build D:\dev\ToolsLibrary\cmake-build-debug-windows --target all
> Scanning dependencies of target ToolsLibrary_random_test
> [  7%] Building CXX object tests/src/CMakeFiles/ToolsLibrary_random_test.dir/random_test.cpp.obj
> random_test.cpp
> D:\dev\ToolsLibrary\tests\src\random_test.cpp(2): error C2230: could not find module 'boost.ut'
> ```

This is to be expected because MSVC requires a `.ixx`/`.cppm` file for the module. Unfortunately trying to make a `ut.cppm`/`ut.ixx` will still fail; trying to do:
```cpp
// ut.ixx
#include <boost/ut.hpp> 
```
for me emits 
```md
1>ut.ixx
1>Compilation...
1>ut.ixx
1>D:\Third-Party\ut\include\boost\ut.hpp(10,18): error C3674: the standard library module 'std' is not found
1>D:\Third-Party\ut\include\boost\ut.hpp(16,10): warning C5244: '#include <ciso646>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <ciso646>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
1>D:\Third-Party\ut\include\boost\ut.hpp(55,10): warning C5244: '#include <array>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <array>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
1>D:\Third-Party\ut\include\boost\ut.hpp(56,10): warning C5244: '#include <iostream>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <iostream>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
1>D:\Third-Party\ut\include\boost\ut.hpp(57,10): warning C5244: '#include <sstream>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <sstream>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
1>D:\Third-Party\ut\include\boost\ut.hpp(58,10): warning C5244: '#include <string_view>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <string_view>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
1>D:\Third-Party\ut\include\boost\ut.hpp(60,10): warning C5244: '#include <vector>' in the purview of module 'boost.ut' appears erroneous.  Consider moving that directive before the module declaration, or replace the textual inclusion with 'import <vector>;'.
1>D:\Third-Party\ut\include\boost\ut.hpp(9,8): message : see module 'boost.ut' declaration
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ option(BOOST_UT_ENABLE_SANITIZERS "Run static analysis" OFF)
 option(BOOST_UT_BUILD_BENCHMARKS "Build the benchmarks" ${MASTER_PROJECT})
 option(BOOST_UT_BUILD_EXAMPLES "Build the examples" ${MASTER_PROJECT})
 option(BOOST_UT_BUILD_TESTS "Build the tests" ${MASTER_PROJECT})
+option(BOOST_UT_DISABLE_MODULE "Disable ut module" OFF)
 
 # ---- Add dependencies via CPM ----
 # see https://github.com/cpm-cmake/CPM.cmake for more info
@@ -129,6 +130,10 @@ endif()
 
 if(BOOST_UT_ENABLE_SANITIZERS)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0 -g -fno-omit-frame-pointer -fsanitize=address -fsanitize=leak -fsanitize=undefined")
+endif()
+
+if(BOOST_UT_DISABLE_MODULE)
+  target_compile_definitions(ut INTERFACE BOOST_UT_DISABLE_MODULE)
 endif()
 
 # Create target Boost::ut and install target

--- a/include/boost/ut.hpp
+++ b/include/boost/ut.hpp
@@ -5,7 +5,7 @@
 // (See accompanying file LICENSE_1_0.txt or copy at
 // http://www.boost.org/LICENSE_1_0.txt)
 //
-#if defined(__cpp_modules)
+#if defined(__cpp_modules) && !defined(BOOST_UT_DISABLE_MODULE)
 export module boost.ut;
 export import std;
 #else
@@ -66,7 +66,7 @@ export import std;
 #include <exception>
 #endif
 
-#if defined(__cpp_modules)
+#if defined(__cpp_modules) && !defined(BOOST_UT_DISABLE_MODULE)
 export namespace boost::inline ext::ut {
 #else
 namespace boost {
@@ -2312,7 +2312,7 @@ using operators::operator/;
 using operators::operator>>;
 }  // namespace v1_1_8
 }  // namespace ut
-#if !defined(__cpp_modules)
+#if !defined(__cpp_modules) || defined(BOOST_UT_DISABLE_MODULE)
 }  // namespace inline ext
 }  // namespace boost
 #endif

